torch_geometric/sampler/base.py

```diff
@@ -105,13 +105,20 @@ class EdgeSamplerInput(CastMixin):
     def __getitem__(self, index: Union[Tensor, Any]) -> 'EdgeSamplerInput':
         if not isinstance(index, Tensor):
             index = torch.tensor(index, dtype=torch.long)
-
+        if self.time is None:
```

Comment:
When is `time` a `tuple`? It should always be of type `Optional[Tensor]`.

Reply:
it is for the mag240m PR i am working on based off of yanbing's original PR
https://github.com/snap-stanford/ogb/pull/449

If i print time before the error:
```
self.time= ('paper', tensor([False, False, False,  ..., False, False, False]))

    self.time['paper'][index] if self.time is not None else None,
TypeError: tuple indices must be integers or slices, not str
```

Reply:
Where and how do you pass `time`. Wouldn't it be sufficient to just pass a `Tensor` there? I don't think there is a need to "fix" this here.

Reply:
basing off of yanbings PR here: https://github.com/snap-stanford/ogb/pull/449
not sure why the time is like that but ill investigate and try to fix in ogb instead of here